docs/source/essentials/normalized-cache.mdx

```diff
@@ -0,0 +1,325 @@
+
+## Data Normalization:
+
+The normalized cache stores objects by ID.
+
+```graphl
+query BookWithAuthorName {
+  favoriteBook {
+    id
+    title
+    author {
+      id
+      name
+    }
+  }
+}
+
+query AuthorById($id: String!) {
+  author(id: $id) {
+      id
+      name
+    }
+  }
+}
+```
+
+In the above example, requesting the author of your favorite book with the `AuthorById` query will return a result from the cache if you requested your favorite book before. This is because the author is stored only once in the cache.
+
+## Storing your data in memory
+
+Apollo Android comes with a `LruNormalizedCache` that will store your data in memory:
+
+<MultiCodeBlock>
+
+```kotlin:title=Kotlin
+// Create a 10MB NormalizedCacheFactory
+val cacheFactory = LruNormalizedCacheFactory(EvictionPolicy.builder().maxSizeBytes(10 * 1024 * 1024).build())
+
+// Build the ApolloClient
+val apolloClient = ApolloClient.builder()
+  .serverUrl("https://...")
+  .normalizedCache(cacheFactory)
+  .build())
+```
+
+```java:title=Java
+// Create a 10MB NormalizedCacheFactory
+NormalizedCacheFactory cacheFactory = new LruNormalizedCacheFactory(EvictionPolicy.builder().maxSizeBytes(10 * 1024 * 1024).build());
+
+// Build the ApolloClient
+ApolloClient apolloClient = ApolloClient.builder()
+  .serverUrl("https://...")
+  .normalizedCache(cacheFactory)
+  .build();
+```
+
+</MultiCodeBlock>
+
+## Persisting your data in a SQLite database
+
+If the amount of data you store becomes too big to fit in memory or if you want your data to persist between app restarts, you can also use a `SqlNormalizedCacheFactory`. A `SqlNormalizedCacheFactory` will store your data in a [SQLDelight](https://github.com/cashapp/sqldelight) database and is defined in a separate dependency:
+
+```kotlin:title=build.gradle.kts
+dependencies {
+  implementation("com.apollographql.apollo:apollo-normalized-cache-sqlite:x.y.z")
+}
+```
+
+**Note:** The `apollo-normalized-cache-sqlite` dependency has Kotlin multiplatform support and has multiple variants (`-jvm`, `-android`, `-ios-arm64`,...). If you are targetting Android and using custom `buildTypes`, you will need to help Gradle resolve the correct artifact by defining [matchingFallbacks](https://developer.android.com/studio/build/dependencies?authuser=2#resolve_matching_errors):
+
+<MultiCodeBlock>
+
+```kotlin:title=build.gradle
+android {
+  buildTypes {
+    create("custom") {
+      // your code...
+      matchingFallbacks = listOf("debug")
+    }
+  }
+}
+```
+
+```groovy:title=build.gradle
+android {
+  buildTypes {
+    custom {
+      // your code...
+      matchingFallbacks = ["debug"]
+    }
+  }
+}
+```
+
+</MultiCodeBlock>
+
+Once the dependency added, you can create the `SqlNormalizedCacheFactory`:
```

Comment:
english nit: Once the dependency *is* added

Reply:
Done, I also took this opportunity to switch to imperative voice, that felt more positive:

```Once the dependency is added, create the `SqlNormalizedCacheFactory`:```